### ISSUE TITLE
You shouldn't be able to remove the mail 3PID from the account if validation is enabled

### ISSUE THREAD

**Issue by maranda** [matrix-org/synapse#10300]
### Description

After registration even if mail validation is required and account validation is enabled you're able to completely remove it without replacing it and having at least one valid e-mail 3PID. This is also a security issue as you could really register an account, log in, remove the 3PID and register another account with the same mail address and a problem in which it can leave accounts in disabled, broken state after they expire and require admin intervention as there's no way for users to receive the renewal tokens.

### Steps to reproduce

- Register an account with mail validation
- Log into the account
- Remove the 3PID from settings general

<!--
Describe how what happens differs from what you expected.

If you can identify any relevant log snippets from _homeserver.log_, please include
those (please be careful to remove any personal or private data). Please surround them with
``` (three backticks, on a line on their own), so that they are formatted legibly.
-->

Ideally if you have mail validation during registration and account validation enabled there should be at least a mail 3PID associated to the account, and Synapse should prevent mail 3PID associated < 1

### Version information

<!-- IMPORTANT: please answer the following questions, to help us narrow down the problem -->

<!-- Was this issue identified on matrix.org or another homeserver? -->
- **Homeserver**: aria-net.org

If not matrix.org:

<!--
 What version of Synapse is running?

You can find the Synapse version with this command:

$ curl http://localhost:8008/_synapse/admin/v1/server_version

(You may need to replace `localhost:8008` if Synapse is not configured to
listen on that port.)
-->
- **Version**: 1.37.1

- **Install method**: package manager
<!-- examples: package manager/git clone/pip  -->

- **Platform**: Ubuntu Server 20.04
<!--
Tell us about the environment in which your homeserver is operating
distro, hardware, if it's running in a vm/container, etc.
-->

**Comment by babolivier** [comment]
I think it'd be better to force users to have at least one email address when registration requires email verification. Forcing that the email provided for registration stays on the account doesn't make a lot of sense for the account's owner in the long run because people tend to change email addresses every now and then, and there's no reason to force a user to keep an old address associated with their Matrix account, especially if there's no technical constraint forcing our hand in this regard (e.g. we don't use email addresses as account identifiers, etc).

**Comment by maranda** [comment]
> I think it'd be better to force users to have at least one email address when registration requires email verification. Forcing that the email provided for registration stays on the account doesn't make a lot of sense for the account's owner in the long run because people tend to change email addresses every now and then, and there's no reason to force a user to keep an old address associated with their Matrix account, especially if there's no technical constraint forcing our hand in this regard (e.g. we don't use email addresses as account identifiers, etc).

Erm, I would like you to tell me where I *ever* mentioned that the same mail address specified in the registration stays there, I just said that if there's either mail validation during registration and / or account validation there should be at least one active mail 3PID associated with the account.

**Comment by maranda** [comment]
Of course they should be able to change it, but they shouldn't be able to remove it leaving the account without an associated mail 3PID in this case.

**Comment by babolivier** [comment]
Right, I misunderstood your comment then, glad we're on the same page.
(and no need to be harsh, I'm just trying to help here :slightly_smiling_face:)

**Comment by maranda** [comment]
(not meant to be harsh as well, sorry it sounded so)

**Comment by babolivier** [comment]
No worries :)

**Comment by Neustradamus** [comment]
@maranda: Good catch!
